x-pack/plugins/task_manager/server/task.ts

```diff
@@ -50,6 +50,7 @@ export type SuccessfulRunResult = {
   state: Record<string, unknown>;
   taskRunError?: DecoratedError;
   skipAttempts?: number;
+  shouldValidate?: boolean;
```

Comment:
nit: would this make more sense in the `FailedRunResult` type?

Reply:
I did it so first but type checker complained about it as we call the same function for both success and failure.